### PR TITLE
#1339F - Quick fix to not show action to Assign on list for Assigner

### PR DESCRIPTION
**Pull request by nmadruga**
Previous problem was this:
Screen Shot 2022-09-08 at 2 43 18 PM

It would show this for the Assigner (or Review with self-assignment rights) regardless of application being Expired (Caduco) or not:

Screen Shot 2022-09-08 at 2 46 06 PM
Screen Shot 2022-09-08 at 2 43 26 PM

### CODE REVIEW

**Comment by CarlosNZ** [comment]
Can you just explain more specifically what situation this handles?

File: database/buildSchema/39_assigner_action_list.sql [hunk 1]
```diff
@@ -18,34 +18,42 @@ CREATE FUNCTION assigner_list (stage_id int, assigner_id int)
...
+            'RE_ASSIGN'
+        WHEN COUNT(DISTINCT (review_assignment.id)) != 0
+            AND assigned_questions_count (application_id, $1, level_number) >= assignable_questions_count (application_id)
+            AND submitted_assigned_questions_count (application_id, $1, level_number) >= assigned_questions_count (application_id, $1, level_number) THEN
+            'ASSIGN_LOCKED'
+        WHEN COUNT(DISTINCT (review_assignment.id)) != 0
+            AND assigned_questions_count (application_id, $1, level_number) < assignable_questions_count (application_id) THEN
+            'ASSIGN'
+        ELSE
+            NULL
```
Comment (CarlosNZ) [inline database/buildSchema/39_assigner_action_list.sql:33]: This is all just formatting changes, right?

File: database/buildSchema/39_assigner_action_list.sql [hunk 2]
```diff
@@ -18,34 +18,42 @@ CREATE FUNCTION assigner_list (stage_id int, assigner_id int)
...
+WHERE
     review_assignment.stage_id = $1
     AND review_assignment_assigner_join.assigner_id = $2
+    AND (
+        SELECT
+            outcome
+        FROM
+            application
+        WHERE
+            id = review_assignment.application_id) = 'PENDING'
```
Comment (CarlosNZ) [inline database/buildSchema/39_assigner_action_list.sql:52]: And this is the only actual change?

**Review by CarlosNZ — COMMENTED** [review]
Just want to check -- this only restricts the *Actions* shown to the assigner, right? Not the actual list of applications returned.

**Comment by nmadruga** [comment]
> Just want to check -- this only restricts the _Actions_ shown to the assigner, right? Not the actual list of applications returned.

Yes, it's just what ACTIONS are available to the assigner for each application returned

File: database/buildSchema/39_assigner_action_list.sql (hunk 2 above)
Comment (nmadruga) [reply database/buildSchema/39_assigner_action_list.sql:52]: Yep

File: database/buildSchema/39_assigner_action_list.sql (hunk 1 above)
Comment (nmadruga) [reply database/buildSchema/39_assigner_action_list.sql:33]: Yea LOL

**Comment by CarlosNZ** [comment]
> Yes, it's just what ACTIONS are available to the assigner for each application returned

Great, all looks good then. (Presume you've tested for unexpected syntax error or whatever)

**Review by CarlosNZ — APPROVED** [review]
LGTM!

**Comment by nmadruga** [comment]
> Can you just explain more specifically what situation this handles?

Have added to description now

**Pull request merged**